crates/sui-storage/src/event_store/sql.rs

```diff
@@ -250,22 +317,34 @@ impl EventStore for SqlEventStore {
                 .bind(event.event.object_id().map(|id| id.to_vec()))
                 .bind(event_to_json(event))
                 .execute(&self.pool)
-                .await?;
+                .await
+                .map_err(convert_sqlx_err)?;
+        }
+
+        // CAS is used to detect any concurrency glitches.  Note that we assume a single writer
+        // append model, which is currently true.  In single writer the CAS should never fail.
+        // We also do this after writing all events, for efficiency.
+        if cur_seq > initial_seq {
+            self.seq_num
```

Comment:
Is `self.seq_num` mainly used to detect multi writer scenarios? Not quite follow what it does :)

Reply:
It is a cache for the latest sequence, so we don't have to pull it out of the DB all the time.